wp-tevko-responsive-images.php

```diff
@@ -48,403 +48,8 @@ function tevkori_get_picturefill() {
-	$size = preg_match( '/size-([^\s|"]+)/i', $image, $match_size ) ? $match_size[1] : false;
-	$width = preg_match( '/ width="([0-9]+)"/', $image, $match_width ) ? (int) $match_width[1] : false;
-	$height = preg_match( '/ height="([0-9]+)"/', $image, $match_height ) ? (int) $match_height[1] : false;
-
-	if ( $id && false === $size ) {
-		$size = array(
-			$width,
-			$height,
-		);
-	}
-
-	/*
-	 * If attempts to parse the size value failed, attempt to use the image
-	 * metadata to match the 'src' against the available sizes for an attachment.
-	 */
-	if ( ! $size && ! empty( $id ) && is_array( $meta = wp_get_attachment_metadata( $id ) ) ) {
-		// Parse the image 'src' value from the 'img' element.
-		$src = preg_match( '/src="([^"]+)"/', $image, $match_src ) ? $match_src[1] : false;
-
-		// Return early if the 'src' value is empty.
-		if ( ! $src ) {
-			return $image;
-		}
-
-		/*
-		 * First, see if the file is the full size image. If not, we loop through
-		 * the intermediate sizes until we find a file that matches.
-		 */
-		$image_filename = wp_basename( $src );
-
-		if ( $image_filename === basename( $meta['file'] ) ) {
-			$size = 'full';
-		} else {
-			foreach( $meta['sizes'] as $image_size => $image_size_data ) {
-				if ( $image_filename === $image_size_data['file'] ) {
-					$size = $image_size;
-					break;
-				}
-			}
-		}
-
-	}
-
-	// If we have an ID and size, try for 'srcset' and 'sizes' and update the markup.
-	if ( $id && $size && $srcset = tevkori_get_srcset( $id, $size ) ) {
-
-		/**
-		 * Pass the 'height' and 'width' to 'tevkori_get_sizes()' to avoid
-		 * recalculating the image size.
-		 */
-		$args = array(
-			'height' => $height,
-			'width'  => $width,
-		);
-		$sizes = tevkori_get_sizes( $id, $size, $args );
-
-		// Format the srcset and sizes string and escape attributes.
-		$srcset_and_sizes = sprintf( ' srcset="%s" sizes="%s"', esc_attr( $srcset ), esc_attr( $sizes) );
-
-		// Add srcset and sizes attributes to the image markup.
-		$image = preg_replace( '/<img ([^>]+)[\s?][\/?]>/', '<img $1' . $srcset_and_sizes . ' />', $image );
-	};
-
-	return $image;
-}
-
-/**
- * Filter to add 'srcset' and 'sizes' attributes to post thumbnails and gallery images.
- *
- * @see wp_get_attachment_image_attributes
- * @return array Attributes for image.
- */
-function tevkori_filter_attachment_image_attributes( $attr, $attachment, $size ) {
-	if ( ! isset( $attr['srcset'] ) ) {
-		$srcset = tevkori_get_srcset( $attachment->ID, $size );
-
-		// Set the 'srcset' attribute if one was returned.
-		if ( $srcset ) {
-			$attr['srcset'] = $srcset;
-
-			if ( ! isset( $attr['sizes'] ) ) {
-				$sizes = tevkori_get_sizes( $attachment->ID, $size );
-
-				// Set the 'sizes' attribute if sizes were returned.
-				if ( $sizes ) {
-					$attr['sizes'] = $sizes;
-				}
-			}
-		}
-	}
-
-	return $attr;
+if ( ! function_exists( 'wp_get_attachment_image_srcset' ) ) {
```

Comment:
I think we should add an explanation in a comment here for people who didn't check the readme/changelog.